src/inference/src/compilation_context.cpp

```diff
@@ -140,6 +140,26 @@ std::string NetworkCompilationContext::computeHash(const std::string& modelName,
     return std::to_string(seed);
 }
 
+std::string NetworkCompilationContext::computeHash(const std::string& modelStr,
+                                                   const ov::Tensor& data,
+                                                   const std::map<std::string, std::string>& compileOptions) {
+    OV_ITT_SCOPE(FIRST_INFERENCE, itt::domains::IE_LT, "NetworkCompilationContext::computeHash - Model Memory");
+    uint64_t seed = 0;
+    // model string
+    seed = hash_combine(seed, modelStr);
+
+    // tensor data
+    seed = hash_combine(seed, data.get_size());
+    std::string dataStr(static_cast<char*>(data.data()));
```

Comment:
I am not sure that it will work in case if some value in data equals to `\0`... Looks like in this case we won't read any next values.

Reply:
Good point! It is a problem if it contains `\0` in the tensor. 

Considered std::hash<std::string> is good way to compute hash value, maybe we can add checking code to check whether there is '\0' (just check string size is ok) and determinate use std::hash<std::string> or fallback to compute hash one value by one value?

Reply:
I confirm that we do encounter the '\0' data in weights data, and the actual string size is less than weights size.

If we compute weights hash one byte by one byte, the time consumption will increase greatly, about 3x~4x than cnn cache or model path cache, that is because weight data size always is very big.

@ilyachur do you have any idea on how to handle such problem? Is there better method to compute weight's hash value?

Reply:
@ilya-lavrenov Do you remember how we solve this issue before?

Reply:
Fix the hash issue to make all weights data can be computed, and the time consumption is below for 2GB weights file:

 
![image](https://user-images.githubusercontent.com/31196718/210938956-ca5f50c7-a199-4b13-a559-4ec2d577fce5.png)